packages/blocks/src/api/raw-handling/html-formatting-remover.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Internal dependencies
+ */
+import { isPhrasingContent } from './phrasing-content';
+
+function getSibling( node, which ) {
+	const sibling = node[ `${ which }Sibling` ];
+
+	if ( sibling ) {
+		return sibling;
+	}
+
+	const { parentNode } = node;
+
+	if ( ! parentNode ) {
+		return;
+	}
+
+	if ( ! isPhrasingContent( parentNode ) ) {
+		return;
+	}
+
+	return getSibling( parentNode, which );
+}
+
+function isFormattingSpace( character ) {
+	return character === ' ' || character === '\t' || character === '\n';
+}
+
+/**
+ * Removes spacing that formats HTML.
+ *
+ * @param {Node} node The node to be processed.
+ * @return {void}
+ */
+export default function( node ) {
+	if ( node.nodeType !== node.TEXT_NODE ) {
+		return;
+	}
+
+	// Ignore pre content.
+	if ( node.parentElement.closest( 'pre' ) ) {
+		return;
+	}
+
+	// First, replace any sequence of HTML formatting space with a single space.
+	let newData = node.data.replace( /[ \n\t]+/g, ' ' );
+
+	// Remove the leading space if the text element is at the start of a block,
+	// is preceded by a line break element, or has a space in the previous
+	// element.
+	if ( newData[ 0 ] === ' ' ) {
+		const previousSibling = getSibling( node, 'previous' );
+
+		if (
+			! previousSibling ||
+			previousSibling.nodeName === 'BR' ||
+			previousSibling.textContent.slice( -1 ) === ' '
```

Comment:
Does the previous element content need to match specifically the space character, or could it be any of the whitespaces (newline, tab, etc). We've normalized to spaces for `newData` on this node, but the same can't be said of `previousSibling`.

**Edit:** I see in the next condition you consider `isFormattingSpace`. For what reason is that not used here as well?

Reply:
Any previous sibling will already have been adjusted, right? But I guess it depends in which direction you loop, so I'll adjust it.

Reply:
Actually since the other block doesn't need the function anymore, it would be easier to leave as is. Is there a test case that's failing?

Reply:
> Actually since the other block doesn't need the function anymore, it would be easier to leave as is. Is there a test case that's failing?

The only thing I could think is if the logic exists here at all to account for spaces, an equivalent test to make sure that it works if that space were in-fact a tab, or a newline.

Reply:
I'm not sure if I follow. :) Could you elaborate?

Reply:
The thinking being: If this logic were removed, I assume there are some tests which start to fail. And for those tests, they fail because of some treatment of `' '`. If we were to change (add a new) the test to use a tab or newline instead, we could verify that it works as expected in those scenarios as well.

I'm not overly concerned about it, since like you mentioned, since it proceeds in order from the previous result, I expect the previous should be normalized.

Reply:
Ah, I understand. The only test that would fail already has a line break in that place: https://github.com/WordPress/gutenberg/pull/17470/files#diff-15a464d6f6d7c014b560b68da1c51c7bR91 (line break after the `strong` tag).